compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1898,7 +1898,7 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
 
         let deps = self
             .tcx
-            .crates(())
+            .crates_including_speculative(())
```

Comment:
Why aren't we encoding just the used crates? The others aren't necessary dependencies